src/gui/src/layoutViewer.cpp

```diff
@@ -474,6 +611,192 @@ void LayoutViewer::updateRubberBandRegion()
+
+  // Cache the search results as we will iterate over the instances
+  // for each layer.
+  std::vector<dbInst*> insts;
+  for (auto& [box, poly, inst] : inst_range) {
+    if (options_->isInstanceVisible(inst)) {
+      insts.push_back(inst);
+      boxes.push_back(box);
+    }
+  }
+
+  // look for edges in metal shapes
+  dbTech* tech = db_->getTech();
+  for (auto layer : tech->getLayers()) {
+    if (!options_->isVisible(layer)) {
+      continue;
+    }
+
+    for (auto inst : insts) {
+      dbMaster* master = inst->getMaster();
+
+      const Boxes* inst_boxes = boxesByLayer(master, layer);
+      if (inst_boxes == nullptr) {
+        continue;
+      }
+      dbTransform inst_xfm;
+      inst->getTransform(inst_xfm);
+
+      if (options_->areObstructionsVisible()) {
+        for (auto& box : inst_boxes->obs) {
+          odb::Rect trans_box(box.left(), box.bottom(), box.right(), box.top());
+          inst_xfm.apply(trans_box);
+          if (trans_box.intersects(search_line)) {
+            boxes.push_back({Search::Point(trans_box.xMin(), trans_box.yMin()), Search::Point(trans_box.xMax(), trans_box.yMax())});
+          }
+        }
+      }
+      for (auto& box : inst_boxes->mterms) {
+        odb::Rect trans_box(box.left(), box.bottom(), box.right(), box.top());
+        inst_xfm.apply(trans_box);
+
+        if (trans_box.intersects(search_line)) {
+          boxes.push_back({Search::Point(trans_box.xMin(), trans_box.yMin()), Search::Point(trans_box.xMax(), trans_box.yMax())});
+        }
+      }
+    }
+
+    auto shapes = search_.searchShapes(layer, search_line.xMin(), search_line.yMin(), search_line.xMax(), search_line.yMax());
+    for (auto& [box, poly, net] : shapes) {
+      if (options_->isNetVisible(net)) {
+        boxes.push_back(box);
+      }
+    }
+
+    if (options_->areFillsVisible()) {
+      auto fills = search_.searchFills(layer, search_line.xMin(), search_line.yMin(), search_line.xMax(), search_line.yMax());
+      for (auto& [box, poly, fill] : fills) {
+        boxes.push_back(box);
+      }
+    }
+
+    if (options_->areObstructionsVisible()) {
+      auto obs = search_.searchObstructions(layer, search_line.xMin(), search_line.yMin(), search_line.xMax(), search_line.yMax());
+      for (auto& [box, poly, ob] : obs) {
+        boxes.push_back(box);
+      }
+    }
+  }
+
+  if (options_->areBlockagesVisible()) {
+    auto blcks = search_.searchBlockages(search_line.xMin(), search_line.yMin(), search_line.xMax(), search_line.yMax());
+    for (auto& [box, poly, blck] : blcks) {
+      boxes.push_back(box);
+    }
+  }
+
+  auto shape_edges = searchNearestEdge(boxes, pt, ok);
+  Edge selected_edge;
+  if (horizontal) {
+    selected_edge = shape_edges.horizontal;
+  } else {
+    selected_edge = shape_edges.vertical;
+  }
+
+  // check if edge should point instead
+  const int point_snap_distance = std::max(10.0 / pixels_per_dbu_, // four pixels
```

Comment:
how is this four pixels?

Reply:
It's not, I think my butter fingers couldn't hit a four pixel target so I increased it and forgot to update comment.